modules/imgcodecs/src/grfmt_png.cpp

```diff
@@ -1440,11 +1463,13 @@ bool PngEncoder::writeanimation(const Animation& animation, const std::vector<in
         if (animation.frames[i].channels() == 3)
             cvtColor(animation.frames[i], tmpframes[i], COLOR_BGR2RGB);
 
+        if (tmpframes[i].depth() != CV_8U)
+            tmpframes[i].convertTo(tmpframes[i], CV_8U, 1.0 / 255);
         apngFrame.setMat(tmpframes[i], animation.durations[i]);
 
         if (i > 0 && !getRect(width, height, frames.back().getPixels(), apngFrame.getPixels(), over1.data(), bpp, rowbytes, 0, 0, 0, 3))
         {
-            frames[i - 1].setDelayNum(frames.back().getDelayNum() + apngFrame.getDelayNum());
+            frames.back().setDelayNum(frames.back().getDelayNum() + apngFrame.getDelayNum());
```

Comment:
@vrabaud @asmorkalov `frames.back().setDelayNum(frames.back().getDelayNum() + apngFrame.getDelayNum());` 
solves the failure on https://github.com/opencv/opencv/actions/runs/12583343235/job/35070690857?pr=26699